src/util/timer.h

```diff
@@ -39,40 +40,45 @@ class Timer {
 // TODO: replace with std::experimental::scope_exit<Timer> once stabilized
 class ScopedTimer {
   public:
-    ScopedTimer(QStringView key,
-            Stat::ComputeFlags compute = kDefaultComputeFlags)
-            : ScopedTimer(key, QStringView(), compute) {
-    }
-    ScopedTimer(QStringView key,
-            int i,
-            Stat::ComputeFlags compute = kDefaultComputeFlags)
-            : ScopedTimer(key,
-                      CmdlineArgs::Instance().getDeveloper()
-                              ? QString::number(i)
-                              : QStringView(),
-                      compute) {
-    }
-
-    ScopedTimer(QStringView key, QStringView arg, Stat::ComputeFlags compute = kDefaultComputeFlags)
+    // Allows the timer to contain a format string which is only assembled
+    // when we're not in `--developer` mode.
+    /// @param compute Flags to use for the Stat::ComputeFlags (can be omitted)
+    /// @param key The format string as QStringLiteral to identify the timer
+    /// @param args The arguments to pass to the format string
+    template<typename T, typename... Ts>
+    ScopedTimer(Stat::ComputeFlags compute, T&& key, Ts&&... args)
             : m_maybeTimer(std::nullopt) {
+        static_assert(std::is_same_v<T, QString>,
+                "only QString is supported as key type. Wrap it in u\"\""
+                "_s or QStringLiteral() "
+                "to avoid runtime UTF-16 conversion.");
         if (!CmdlineArgs::Instance().getDeveloper()) {
-            return;
+            return; // leave timer in cancelled state
         }
-#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
-        QString strKey = arg.isEmpty() ? key.toString() : key.arg(arg);
-#else
-        QString strKey = arg.isEmpty() ? key.toString() : key.toString().arg(arg);
-#endif
-        m_maybeTimer = std::make_optional<Timer>(std::move(strKey), compute);
+        //
+        auto assembledKey = QString(std::forward<T>(key));
```

Comment:
This one is unfortunately not for free, because the compiler decides to create a local String copy: 
https://godbolt.org/z/67M4MTc6j
Here is the `const QString&``version:  
https://godbolt.org/z/9WbMedoej

Reply:
But in both cases, it still holds that `capacity() == 0`, thus the copy is not heap-allocated, so I don't really see a problem. Anyways, wdyt of the fixup. It avoids the construction of the temporary by use of an IILE.